pkg/ipam/release.go

```diff
@@ -111,7 +111,7 @@ func (i *ipam) releaseForAllNICs(ctx context.Context, uid, nic string, endpoint
 	// Check whether an StatefulSet needs to release its currently allocated IP addresses.
```

Comment:
// compare the annotation between the deleting pod and statefulset yaml , to check whether the annotation is changed